_data/doc_categories.yml

```diff
@@ -1,14 +1,20 @@
-# The order of the categories in this file (first-to-last) determines their
-# order in the UI (left-to-right)
+latest:
 - title: Concepts
-  href: docs/concepts/
+  href: docs/$$VERSION$$/concepts/
 - title: Setup
-  href: docs/setup/
+  href: docs/$$VERSION$$/setup/
 - title: Configure
-  href: docs/configure/
+  href: docs/$$VERSION$$/configure/
 - title: Use
-  href: docs/use/
+  href: docs/$$VERSION$$/use/
 - title: Develop
-  href: docs/develop/
+  href: docs/$$VERSION$$/develop/
 - title: FAQ
-  href: docs/faq/
+  href: docs/$$VERSION$$/faq/
+v1.1:
```

Comment:
Will the script create these heading for us?

Reply:
Yes, a snapshot of the headings is created such that it pulls whatever is from `latest` and creates a new entry for the new version. So, the latest headings have to be kept inline with the documentation (as was done before).

So, in the case that we have:
```yaml
latest:
- title: <example-title>
  href: <example-url>
v2.0:
- title: Concepts
  href: docs/$$VERSION$$/concepts/
- title: Setup
  href: docs/$$VERSION$$/setup/
- title: Configure
  href: docs/$$VERSION$$/configure/
- title: Use
  href: docs/$$VERSION$$/use/
- title: Develop
  href: docs/$$VERSION$$/develop/
- title: FAQ
  href: docs/$$VERSION$$/faq/
v1.1:
- href: docs/$$VERSION$$/quickstarts/
  title: Quickstarts
- href: docs/$$VERSION$$/howto/
  title: Howtos
- href: docs/$$VERSION$$/guides/
  title: Guides
```
when a new version is generated for say `v2.1` it will become:
```yaml
latest:
- title: <example-title>
  href: <example-url>
v2.1:
- title: <example-title>
  href: <example-url>
v2.0:
- title: Concepts
  href: docs/$$VERSION$$/concepts/
- title: Setup
  href: docs/$$VERSION$$/setup/
- title: Configure
  href: docs/$$VERSION$$/configure/
- title: Use
  href: docs/$$VERSION$$/use/
- title: Develop
  href: docs/$$VERSION$$/develop/
- title: FAQ
  href: docs/$$VERSION$$/faq/
v1.1:
- href: docs/$$VERSION$$/quickstarts/
  title: Quickstarts
- href: docs/$$VERSION$$/howto/
  title: Howtos
- href: docs/$$VERSION$$/guides/
  title: Guides
```

Reply:
It should be noted, that if the categories are changed then they must be updated in the `_config.yml` as well such that the defaults are set properly. When creating a new version, the `./scripts/create_new_version_from_latest.sh` pulls the version config defaults from `./scripts/data/version_config_defaults.yml` and adds them to `_config.yml`.

So in the example above, where the categories are modified in latest, they should also be modified in `./scripts/data/version_config_defaults.yml` to look like:
```yaml
  - scope:
      path: _docs/$$VERSION$$
    values:
      category: Overview ($$VERSION$$)
      layout: docs
  - scope:
      path: <example-url>
    values:
      category: <example-title> ($$VERSION$$)
```
This is as it was before. The `_config.yml` and `_data/doc_categories.yml` have to be kept in-sync for categories to work properly. There's no easy way to automated detection of new categories because the titles and url mappings have to be written by the documentation writers -- or more specifically whomever is responsible for the information architecture.

Reply:
Thanks. Can we just be sure this is part of the developer docs? Also can we be sure to consistently added a comment to the head of every file that it's auto-generated?

Reply:
Absolutely, and sure thing. Just a note re: `_data/doc_categories.yml`, while a portion of it is auto-generated, the `latest:` key is meant to be maintained by documentation authors. Therefore it's a "auto-generate" but "do modify", therefore, I will note as such in the file head.